packages/database/src/migrations/20210913042439-Add7DayPositivityMapOverlayCovidFiji-modifies-data.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+import {
+  insertObject,
+  generateId,
+  codeToId,
+  arrayToDbString,
+  findSingleRecord,
+  findSingleRecordBySql,
+} from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const getFrontEndConfig = measureLevel => ({
+  scaleType: 'performanceDesc',
+  valueType: 'percentage',
+  displayType: 'shaded-spectrum',
+  customLabel: '% Positivity rate',
+  scaleBounds: {
+    left: {
+      max: 0,
+      min: 0,
+    },
+  },
+  measureLevel,
+  measureConfig: {
+    $all: {
+      type: 'popup-only',
+      hideFromLegend: true,
+    },
+  },
+  hideByDefault: {
+    null: true,
+  },
+});
+
+const getReportConfig = measureLevel => ({
+  fetch: {
+    aggregations: [
+      {
+        type: 'SUM_PER_PERIOD_PER_ORG_GROUP',
+        config: {
+          dataSourceEntityType: 'facility',
+          aggregationEntityType: measureLevel === 'District' ? 'district' : 'sub_district',
+        },
+      },
+    ],
+    dataElements: ['COVID_FJ_7_Day_Rolling_Pos_Tests', 'COVID_FJ_7_Day_Rolling_Num_Tests'],
+  },
+  transform: [
+    'keyValueByDataElementName',
+    {
+      transform: 'updateColumns',
+      insert: {
+        Pos_Tests: '=$COVID_FJ_7_Day_Rolling_Pos_Tests',
+        Num_Tests: '=$COVID_FJ_7_Day_Rolling_Num_Tests',
+      },
+      include: ['organisationUnit'],
+    },
+    {
+      transform: 'mergeRows',
+      groupBy: 'organisationUnit',
+      using: 'sum',
+    },
+    {
+      transform: 'excludeRows',
+      where: '=not(exists($Num_Tests)) or $Num_Tests <= 0 or not(exists($Pos_Tests))',
+    },
+    {
+      transform: 'updateColumns',
+      insert: {
+        value: '=divide($Pos_Tests,$Num_Tests)',
+        'Positive tests': '=$Pos_Tests',
+        organisationUnitCode: '=$organisationUnit',
+        'Total tests': '=$Num_Tests',
+      },
+      exclude: '*',
+    },
+  ],
+});
+
+const FIJI_HIERARCHY_NAMES = {
+  District: 'Division',
+  SubDistrict: 'Sub-division',
+};
+const getConfig = measureLevel => ({
+  code: `FJ_Covid_7_Day_Positivity_Rate_Overlay_${FIJI_HIERARCHY_NAMES[measureLevel]}`,
+  name: `7 day % positivity rate (${FIJI_HIERARCHY_NAMES[measureLevel]})`,
+  frontEndConfig: getFrontEndConfig(measureLevel),
+  reportConfig: getReportConfig(measureLevel),
+  userGroup: 'BES Admin',
```

Comment:
```suggestion
  userGroup: 'Public',
```
Not sure about this, just judging by the ticket? But I doubt it's meant to be BES Admin...

Reply:
You are totally right, but all of these vizzes are to be released as BES Admin and then changed post-release to Public when ready.

Reply:
Ahh interesting, sounds good 👍